### ISSUE TITLE
Could `--max-records-in-ram` be a top-level param?

### ISSUE THREAD

**Issue by mjhipp** [fulcrumgenomics/fgbio#784]
Similar to `--compression` and `--sam-validation-stringency`.

Currently, `--max-records-in-ram` is an option in some tools, such as `SortBam`. 

Other tools that also sort, such as `ClipBam`, do not have this option, and will always use the default, which is `1M` records. If I had less RAM or more RAM, it would be nice to make other tools that use `Bams.sorter()` use a specified amount of records.

**Comment by nh13** [comment]
Sgtm, @tfenne

**Comment by tfenne** [comment]
Sure, I see no reason not to have a global way to set the default.  Will probably want to make that number accessible to tools too though as some will do things like divide it into two or three and generate multiple sorters/caches, so it'd be more than _just_ changing the default on Sorter/SamWriter.

**Comment by clintval** [comment]
I think my opinion is a minority one, but I've never liked the GATK/Picard/fgbio common option strategy.

There are many tools in fgbio where very few of the common options are actually relevant, yet these options still appear in the help string of the tool. Additionally, some tools never use any of the common options and it [can make for confusing UX](https://twitter.com/clintcodesbio/status/1328511910166327297) (seriously, try `picard NormalizeFasta --help` and find out).

Additionally, there can be ambiguity with how these options affect the tool at runtime and `--max-records-in-ram` is tricky, there are so many "records" this could presumably mean.

If I could start from scratch I would use a trait system on the CLI tools to enforce common options at the sake of some repeated code. Those traits could then override any global setting in the trait body, or the parameter could be used directly in the tool itself since it would also be a class parameter.

**Comment by tfenne** [comment]
That's fair @clintval - though I do think that at least `fgbio` is better than GATK/picard in this case in that the options are separated out and are clearly "toolkit" level options, vs. in Picard/GATK them being mixed into the options for the specific tool.

**Comment by mjhipp** [comment]
I guess what I am really requesting is that more tools that could benefit from it make the `--max-records-in-ram` option available. I just figured the most logical implementation was to make it a top-level param.

**Comment by clintval** [comment]
I agree fgbio is better by partitioning them! @mjhipp I'm cool with another top-level param, I think there is a precedent for it. Simply wanted to churn the pot with a warning about how this has played out in other UIs so that we can be more careful!
